sdk/servicebus/service-bus/src/serviceBusMessage.ts

```diff
@@ -635,13 +640,19 @@ export function fromRheaMessage(
   rawMessage.bodyType = bodyType;
 
   if (rawMessage.applicationProperties) {
-    rawMessage.applicationProperties = convertDatesToNumbers(rawMessage.applicationProperties);
+    rawMessage.applicationProperties = keepDateType
+      ? rawMessage.applicationProperties
+      : convertDatesToNumbers(rawMessage.applicationProperties);
   }
   if (rawMessage.deliveryAnnotations) {
-    rawMessage.deliveryAnnotations = convertDatesToNumbers(rawMessage.deliveryAnnotations);
+    rawMessage.deliveryAnnotations = keepDateType
```

Comment:
maybe I should limit this option to `applicationProperties`.  Then `preserveDateTypeInAppProperties` option name would be better (but too long...)